x-pack/plugins/alerting/server/rules_client/lib/retry_if_bulk_disable_conflicts.ts

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import pMap from 'p-map';
+import { chunk } from 'lodash';
+import { KueryNode } from '@kbn/es-query';
+import { Logger, SavedObjectsBulkUpdateObject } from '@kbn/core/server';
+import { convertRuleIdsToKueryNode } from '../../lib';
+import { BulkOperationError } from '../rules_client';
+import { waitBeforeNextRetry, RETRY_IF_CONFLICTS_ATTEMPTS } from './wait_before_next_retry';
+import { RawRule } from '../../types';
+
+const MAX_RULES_IDS_IN_RETRY = 1000;
+
+export type BulkDisableOperation = (filter: KueryNode | null) => Promise<{
+  errors: BulkOperationError[];
+  rules: Array<SavedObjectsBulkUpdateObject<RawRule>>;
+  taskIdsToDisable: string[];
+  taskIdsToDelete: string[];
+}>;
+
+interface ReturnRetry {
+  errors: BulkOperationError[];
+  rules: Array<SavedObjectsBulkUpdateObject<RawRule>>;
+  taskIdsToDisable: string[];
+  taskIdsToDelete: string[];
+}
+
+/**
+ * Retries BulkEnable requests
+ * If in response are presents conflicted savedObjects(409 statusCode), this util constructs filter with failed SO ids and retries bulkEnable operation until
+ * all SO updated or number of retries exceeded
+ * @param logger
+ * @param bulkEnableOperation
```

Comment:
Nit: update comments so they say `disable` instead of `enable`

Reply:
Fixed: dc4edab775fdec31069932f401ad880fd3234118